jax/_src/lax/qdwh.py

```diff
@@ -123,7 +123,7 @@ def _qdwh(x, m, n, is_hermitian, max_iterations, eps):
   if eps is None:
     eps = float(jnp.finfo(x.dtype).eps)
   alpha = (jnp.sqrt(jnp.linalg.norm(x, ord=1)) *
-           jnp.sqrt(jnp.linalg.norm(x, ord=jnp.inf)))
+           jnp.sqrt(jnp.linalg.norm(x, ord=jnp.inf))).astype(x.dtype)
   l = eps
 
   u = x / alpha
```

Comment:
I'm just reflecting on this. Do we really want to force a cast when doing a `complex / real` division where the `real` type is the same on one of the complex element types?

Reply:
That's a good question – I debated that in https://github.com/google/jax/pull/10824, but landed on the fact that when people say they want `strict` dtype promotion, we should probably avoid any implicit promotion rather than trying to guess which promotions are OK.

Reply:
(I think I even posed this exact question in our x64 chat at some point)

Reply:
I'm going to leave as-is for now; we can always make this less restrictive in the future if we wish.